system_setup/server/controllers/configure.py

```diff
@@ -40,89 +41,189 @@ def __init__(self, app):
+            command line to output into a specified directory.
+            Additionally, indicates success by returning True
+            as the second element of the tuple. """
+
+        cmd = "usr/sbin/locale-gen"
+        if self.app.opts.dry_run is False or self.app.opts.dry_run is None:
+            return (os.path.join("/", cmd), True)
+
+        outDir = os.path.join(self.model.root, "usr/lib/locale")
+        usrSbinDir = os.path.join(self.model.root, "usr/sbin")
+        os.makedirs(outDir, exist_ok=True)
+        os.makedirs(usrSbinDir, exist_ok=True)
+        cmdFile = os.path.join(self.model.root, cmd)
+        shutil.copy(os.path.join("/", cmd), cmdFile)
+        # Supply LC_* definition files to avoid complains from localedef.
+        shutil.copytree("/usr/lib/locale/C.UTF-8/", outDir,
+                        dirs_exist_ok=True)
 
-        fileContents: str
-        locGenNeedsWrite = False
-        with open(localeGenPath, "r") as f:
-            fileContents = f.read()
-            lineFound = fileContents.find(lang)
-            if lineFound == -1:
-                # An unsupported locale coming from our UI is a bug
-                log.error("Selected language %s not supported.", lang)
-                return
-
-            pattern = r'#+\s*({}.*)'.format(lang)
-            commented = re.compile(pattern)
-            (fileContents, n) = commented.subn(r'\1', fileContents, count=1)
-            locGenNeedsWrite = n == 1
-
-        if locGenNeedsWrite:
-            try:
-                with open(localeGenPath, "wt") as f:
-                    f.write(fileContents)
-            except IOError as err:
-                log.error("Failed to modify %s file. %s", localeGenPath, err)
-
-    async def _install_check_lang_support_packages(self, lang, env):
+        try:
+            # Altering locale-gen script to output to the desired folder.
+            with open(cmdFile, "r+") as f:
+                script = f.read()
+                pattern = r'(localedef .+) (\|\| :; \\)'
+                localeDefRe = re.compile(pattern)
+                replacement = r'\1 "{}/" \2'.format(outDir)
+                (fileContents, n) = localeDefRe.subn(replacement, script,
+                                                     count=1)
+                if n != 1:
+                    log.error("locale-gen script contents were unexpected."
+                              " Aborting mock creation")
+                    return ("", False)
+
+                f.seek(0)
+                f.write(fileContents)
+
+            return (cmdFile, True)
+
+        except IOError as err:
+            log.error("Failed to modify %s file. %s", cmdFile, err)
+            return ("", False)
+
+    def __update_locale_cmd(self, lang) -> list[str]:
+        """ Adds mocking cli to update-locale if in dry-run. """
+        updateLocCmd = ["update-locale", "LANG={}".format(lang)]
+        if self.app.opts.dry_run:
```

Comment:
can I nitpick asking for an early return here? (feel free to ignore, this is really some personal preference)